code/hdb/startup.q

```diff
@@ -1,4 +1,55 @@
 
+//called after new HDB cluster finishes loading all dependencies
+//triggers RDB reload instead of the HDB and removes old HDB cluster if it is a replacement
+.hdb.startup:{
+  gws:exec w from .servers.SERVERS where proctype = `gateway, .dotz.liveh w;
+  if[not count gws; :()]; // TODO : Handle case when none
+
+  // block queries to new hdb
+  gws @\: (`.gw.setserveractiveflag;.proc.procname;0b);
+  // wait for status change
+  .finspace.checkstatus[(.finspace.getcluster;.proc.procname);("RUNNING";"FAILURE");00:01;0D01];
+
+  // identify the row and retry connection
+  {[h;tgt]
+     tgtidx:first @[h;({exec i from .servers.SERVERS where procname=x};tgt);()];
+     if[null tgtidx; :()];
+     doretry:@[h;({null .servers.SERVERS[x][`w]};tgtidx);1b];
+     .lg.o[`hdbstartup;"will force retry row ",(-3!tgtidx)," in servers.SERVERS for gw handle ",(-3!h)," : ",-3!doretry];
+     if[doretry; h(`.servers.retryrows;tgtidx)];
+   }[;.proc.procname] each gws;
+
+  rdbs:select procname,w from .servers.SERVERS where proctype = `rdb, .dotz.liveh w;
+  // block queries to the rdbs
+  rdbnames:exec procname from rdbs;
+  gws @\: (`.gw.setserveractiveflag;rdbnames;0b);
+
+  // clear data from rdbs
+  .lg.o[`hdbstartup;"trigger rdbs to remove rows found at eod"];
+  rdbhandles:exec w from rdbs;
+  func:{ if[`eodtabcount in key .rdb; .rdb.reload[x]] };
+  rdbhandles @\: (func;.z.d);
+  .lg.o[`hdbstartup;"finished signaling rdbs to drop rows at time of eod"];
+
+  // block queries to old hdb and begin shutdown process
+  if[count deleteme:first .proc.params[`replaceProc];
+      .lg.o[`hdbstartup;"sending signal to delete the cluster named ",deleteme];
+      gws @\: (`.finspace.unregisterfromgw;enlist `$deleteme);
+    ];
+
+  // enable queries to the rdbs
+  gws @\: (`.gw.setserveractiveflag;rdbnames;1b);
+  // enable queries to new hdb
+  gws @\: (`.gw.setserveractiveflag;.proc.procname;1b);
```

Comment:
I think we should be doing this in a single call rather than two calls

```suggestion
  // enable queries to the rdbs & new hdb
  gws @\: (`.gw.setserveractiveflag;.proc.procname,rdbnames;1b);
```

We possibly also have an issue here in that the unregister of the old HDB & the marking as active of the new HDB and RDB are two separate calls - I think a query could come in to gateway between these and fail as no servers are "active" in gateway? It would be better if we can do this all in a single call to GW.